components/search_results_item.jsx

```diff
@@ -163,7 +165,7 @@ export default class SearchResultsItem extends React.Component {
         const channel = this.props.channel;
         const user = this.props.user || {};
         const post = this.props.post;
-
+        const commentCountForPost = this.getCommentCountForPost(store.getState(), {post});
```

Comment:
0/5 but I thought we should avoid such call in render function.

Reply:
i am 0/5 on this too. Ideally, if this is a redux component I would have preferred this as props for the component.
This should be fine given memorisation in constructor works fine.

Reply:
Preferably I'd like this to be migrated to be redux-friendly but that's a lot of work out-of-scope of the PR. As is the comment count changing won't cause a render, so the search result could have an outdated comment count compared to what's in the main channel view.

If you don't want to fully convert to redux-friendly, what about just partially converting only to pass in the comment count as a prop?

Reply:
Ok, let me see. I will change this to use redux-friendly component or add the commentCount prop. But adding comment count prop means adding a new selector for commentCounts array based on ID's in search results component.

Reply:
I think there is already a selector you can use for that https://github.com/mattermost/mattermost-redux/blob/master/src/selectors/entities/posts.js#L291

Reply:
@jwilander I thought to change `search_results` component to have the selector till we have `search_results_item` as react-redux component and hence array of comment counts. I can make this a connected component and add this selector but felt odd as I have to give PR without actually migrating and just one extra prop. Gimme couple of days i most likely will change this to redux-component and add test cases